textpipe/doc.py

```diff
@@ -60,18 +82,26 @@ def detect_language(self, hint_language=None):
         hint_language: language you expect your text to be
 
         Returns:
+        is_reliable: is the top language is much better than 2nd best language?
         language: 2-letter code for the language of the text
 
         >>> from textpipe.doc import Doc
         >>> doc = Doc('Test')
-        >>> doc.language
-        'en'
+        >>> doc.detect_language()
+        (True, 'en')
         >>> doc.detect_language('nl')
-        'nl'
+        (True, 'nl')
+        >>> Doc('...').detect_language()
+        (False, 'un')
         """
-        _, _, best_guesses = cld2.detect(self.clean, hintLanguage=hint_language,
-                                         bestEffort=True)
-        return best_guesses[0][1]
+        is_reliable, _, best_guesses = cld2.detect(self.clean,
+                                                   hintLanguage=hint_language,
+                                                   bestEffort=True)
+
+        if len(best_guesses) == 0 or len(best_guesses[0]) != 4 or best_guesses[0][1] == 'un':
+            return False, 'un'
```

Comment:
Not entirely sure if it is possible that `cld2.detect` would return the option of `'un'` as `best_guess` and `is_reliable == True`, but with this change it is not possible for sure.